controller/tests/controller_server/csi_controller_server_test.py

```diff
@@ -16,100 +16,126 @@
 from controller.tests import utils
 
 
-class AbstractControllerTest(unittest.TestCase):
+class BaseControllerSetUp(unittest.TestCase):
 
     def setUp(self):
         patch("controller.array_action.array_mediator_xiv.XIVArrayMediator._connect").start()
         detect_array_type_patcher = patch("controller.controller_server.csi_controller_server.detect_array_type")
         self.detect_array_type = detect_array_type_patcher.start()
         self.detect_array_type.return_value = "a9k"
         self.addCleanup(detect_array_type_patcher.stop)
+        self.fqdn = "fqdn"
+        self.mediator = XIVArrayMediator("user", "password", self.fqdn)
+        self.mediator.client = Mock()
+
+        self.storage_agent = MagicMock()
+        self.storage_agent.get_mediator.return_value.__enter__.return_value = self.mediator
+
+        self.servicer = ControllerServicer(self.fqdn)
+
         self.request = Mock()
         self.request.secrets = {"username": "user", "password": "pass", "management_address": "mg"}
 
+        self.request.parameters = {}
+        self.request.volume_context = {}
+        caps = utils.get_mock_volume_capability_object()

```

Comment:
shouldn't `caps` be `volume_capability` (singular)?

Reply:
done

Reply:
**same for [the rest of the places](https://github.com/IBM/ibm-block-csi-driver/blob/d11ee357262e35fce6936539ecf391d32f504209/controller/tests/controller_server/utils_test.py#L57)**